File: backend/app/schemas/session.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime

class SessionBase(BaseModel):
    name: str
    
class SpeakerCreate(SessionBase):
    pass

class SpeakerUpdate(SessionBase):
    pass

class SpeakerResponse(SessionBase):
    id: int
    
class SessionCreate(SessionBase):
    description: str
    duration: int
    date_start: datetime
    speaker_id: int
    
class SessionUpdate(SessionBase):
    description: str
    duration: int
    date_start: datetime
    speaker_id: int
    
class SessionResponse(SessionBase):
    id: int
    description: str
    duration: int
    date_start: datetime
    speaker_id: int
    speaker: SpeakerResponse
    
    model_config = ConfigDict(from_attributes=True)

File: frontend/src/app/session-managment/session.ts
import { Speaker } from "./speaker";

export class Session {
  id: number;
  name: string;
  description: string;
  date_start: string;
  duration: number;
  speaker_id: number;
  speaker: Speaker;

  constructor(
    id: number,
    name: string,
    description: string,
    date_start: string,
    duration: number,
    speaker_id: number,
    speaker: Speaker
  ) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.date_start = date_start;
    this.duration = duration;
    this.speaker_id = speaker_id;
    this.speaker = speaker;
  }
}
